dev/hal/src/Buzzer.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file Buzzer.cpp
+ * @author
+ * @brief
+ * @version 0.1
+ * @date 2024-07-7
+ *
+ * @copyright Copyright (c) 2024
+ *
+ */
+#include "mcal/inc/stm32f103xx.h"
+#include "mcal/inc/Pin.h"
+#include "mcal/inc/Gpio.h"
+#include "hal/inc/Buzzer.h"
+
+using namespace stm32::dev::mcal::pin;
+using namespace stm32::dev::mcal::gpio;
+using namespace stm32::dev::hal::buzzer;
+
+Buzzer::Buzzer(const Pin buzzerPin, const ConnectionType connectionType)
+    : buzzerPin_(buzzerPin), connectionType_(connectionType) {
+        Gpio::SetOutputMode(buzzerPin_, OutputMode::kPushPull_2MHZ);
```

Comment:
What about using the mode as stored in `pin` object instead of forcing user to use `kPushPull_2MHZ` and we can have assert if the `pin` is configured as input pin